source/m2mnsdlinterface.cpp

```diff
@@ -1649,6 +1641,25 @@ void M2MNsdlInterface::handle_bootstrap_error()
     _observer.bootstrap_error();
 }
 
+void M2MNsdlInterface::set_connection_handler(M2MConnectionHandler *connection_handler)
```

Comment:
Probably was thinking backward compatibility by extending rather than changing the constructor but now that I think about it I think it would be better to do it that way.